### PR TITLE
DAOS-2478 duns: switch to only Lustre build deps mode

### PR DESCRIPTION
**Pull request by bfaccini**
As due to licensing it is not possible to copy/use Lustre source
code, change from full (build/install/run) no Lustre deps mechanics
to build deps only. This means that to allow for Lustre specifics
to be enabled, libduns must be built on a platform where
lustre[-client] distro is installed.

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>
Change-Id: I94dd4b243fe783d08b25e0526cf1a5a41247233a

### CODE REVIEW

**Comment by bfaccini** [comment]
> Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-1429/7/
> Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules
> 
> FYI: Errors found in lines not modified in the patch:
> 
> [src/client/dfs/SConscript:10](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L10):
> (pylint-pointless-string-statement) String statement has no effect
> 
> [src/client/dfs/SConscript:12](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L12):
> (pylint-invalid-name) Invalid variable name "f"
> 
> [src/client/dfs/SConscript:23](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L23):
> (pylint-invalid-name) Invalid variable name "ts"

Do I need to fix these old cosmetic errors not from my changes ??

File: src/client/dfs/SConscript [hunk 1]
```diff
@@ -14,28 +14,31 @@ def scons():
...
+        if distro in ['centos', 'rhel', 'fedora', 'opensuse']:
             import rpm
+            from pkg_resources import parse_version
+
             ts = rpm.TransactionSet()
             headers = ts.dbMatch('name', 'lustre')
             if not headers:
-                print("No package lustre found.")
+                headers = ts.dbMatch('name', 'lustre-client')
+            if headers and parse_version(headers['version']) >= parse_version('2.12.57'):
```
Comment (jolivier23) [inline src/client/dfs/SConscript]: One possible alternative here is to use SCons configure with perhaps a custom header that fails to compile if the check fails

config = Configure(env)
has_support = config.CheckHeader(header_name)
config.Finish()

**Comment by jolivier23** [comment]
> > Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-1429/7/
> > Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules
> > FYI: Errors found in lines not modified in the patch:
> > [src/client/dfs/SConscript:10](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L10):
> > (pylint-pointless-string-statement) String statement has no effect
> > [src/client/dfs/SConscript:12](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L12):
> > (pylint-invalid-name) Invalid variable name "f"
> > [src/client/dfs/SConscript:23](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L23):
> > (pylint-invalid-name) Invalid variable name "ts"
> 
> Do I need to fix these old cosmetic errors not from my changes ??

generally, no.  But should try to avoid adding new python ones where reasonable.

File: src/client/dfs/SConscript (hunk 1 above)
Comment (bfaccini) [reply src/client/dfs/SConscript]: > One possible alternative here is to use SCons configure with perhaps a custom header that fails to compile if the check fails
> 
> config = Configure(env)
> has_support = config.CheckHeader(header_name)
> config.Finish()

Thanks for the hint Jeff, but by header do you mean an include file name ?

**Comment by bfaccini** [comment]
> > > Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-1429/7/
> > > Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules
> > > FYI: Errors found in lines not modified in the patch:
> > > [src/client/dfs/SConscript:10](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L10):
> > > (pylint-pointless-string-statement) String statement has no effect
> > > [src/client/dfs/SConscript:12](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L12):
> > > (pylint-invalid-name) Invalid variable name "f"
> > > [src/client/dfs/SConscript:23](https://github.com/daos-stack/daos/blob/163e04fa05ed40e652ba0568f0b017e63405960c/src/client/dfs/SConscript#L23):
> > > (pylint-invalid-name) Invalid variable name "ts"
> > 
> > 
> > Do I need to fix these old cosmetic errors not from my changes ??
> 
> generally, no. But should try to avoid adding new python ones where reasonable.

For sure !!!

**Pull request merged**